pkgs/development/python-modules/cupy/default.nix

```diff
@@ -18,8 +18,11 @@ buildPythonPackage rec {
     mock
   ];
 
-  propagatedBuildInputs = [
+  propagatedNativeBuildInputs = [
```

Comment:
if it's just for nvcc to be on the path, this makes sense. Is there more that cudatoolkit provides?

My main concern is for cross compilation, in which it will get a binary of the wrong architecture.

However, seeing as  `linuxPackages.nvidia_x11` gets propagated as well, this expression probably isn't very cross compilation friendly

Reply:
`cudatoolkit` has both `nvcc` used at build time and libraries linked at runtime. I have no idea whether it might support cross compilation. But the question is entirely moot at this time, as we only package it for Linux x86-64.

Reply:
> `cudatoolkit` has both `nvcc` used at build time and libraries linked at runtime.

If `cudatoolkit` doesn't contain any python libraries then it should not be in `propagated`. Now for libraries to link it should be in `buildInputs`. I believe, there is some strange thing that may be specific to python derivations, where binaries may not be visible unless they are also in `nativeBuildInputs` [1]. I never dug into this, but it seems like a bug in nixpkgs, so you might need to also have it in `nativeBuildInputs` as a workaround.

[1] https://github.com/NixOS/nixpkgs/pull/64047#discussion_r299278096